src/main/java/net/fabricmc/loader/impl/discovery/RuntimeModRemapper.java

```diff
@@ -107,13 +101,50 @@ public static void remap(Collection<ModCandidate> modCandidates, Path tmpDir, Pa
 					info.inputIsTemp = true;
 				}
 
+				info.outputPath = outputDir.resolve(mod.getDefaultFileName());
+				Files.deleteIfExists(info.outputPath);
+
+				String accessWidener = mod.getMetadata().getAccessWidener();
+
+				if (accessWidener != null) {
+					info.accessWidenerPath = accessWidener;
+
+					try (FileSystemUtil.FileSystemDelegate jarFs = FileSystemUtil.getJarFileSystem(info.inputPath, false)) {
+						FileSystem fs = jarFs.get();
+						info.sourceAccessWidener = Files.readAllBytes(fs.getPath(accessWidener));
+					} catch (Throwable t) {
+						throw new RuntimeException("Error reading access widener for mod '" +mod.getId()+ "'!", t);
+					}
+
+					new AccessWidenerReader(mergedAccessWidener).read(info.sourceAccessWidener);
```

Comment:
Nit: Can the `AccessWidenerReader` be reused between mods, I dont think there is a need to create a new one each time.